src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,117 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class RegistrationServiceImplTest {
+    static StorageDao storageDao;
+    static RegistrationService registrationService;
+    static User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+        user = new User();
+    }
+
+    @BeforeEach
+    void setUp(){
+        user.setId(123456789L);
+        user.setLogin("userLogin");
+        user.setPassword("validPassword");
+        user.setAge(20);
+    }
+
+    @Test
+    void register_userIsNull_notOk() {
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_ageIsNull_notOk() {
+        user.setAge(null);
+        System.out.println(user.toString());
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_ageIsLessThan18_notOk() {
```

Comment:
I would not write specific numbers in the name of the method. Maybe someday we want to change the values MIN_AGE and MIN_COUNT_CHARACTER
Better register_ageIsLessThan_MIN_AGE_notOk